src/gd.c

```diff
@@ -2201,9 +2201,9 @@ static void _gdImageFillTiled(gdImagePtr im, int x, int y, int nc)
 	FILL_PUSH(y+1, x, x, -1);
 	while (sp>stack) {
 		FILL_POP(y, x1, x2, dy);
-		for (x=x1; x>=0 && (!pts[y + x*wx2] && gdImageGetPixel(im,x,y)==oc); x--) {
```

Comment:
you are correct that we want [y + x*im->sy]

i wasn't suggesting trying to allocate a 2d array since it's too expensive.  a local macro might not be so bad if it was localized to the func and you only pass in (x,y).  then again, doing a sx*sy*1 byte array when we just care about bools is a bit expensive.  we could add macros to handle setting bits instead which would cut the memory requirement to 1/8th.  if you're interested in that, it can be a follow up PR :).